File: src/client/acontext-py/src/acontext/types/skill.py
"""Type definitions for skill resources."""

from typing import Any

from pydantic import BaseModel, Field

from .common import FileContent


class Skill(BaseModel):
    """Skill model representing an agent skill resource."""

    id: str = Field(..., description="Skill UUID")
    project_id: str = Field(..., description="Project UUID")
    name: str = Field(..., description="Skill name (unique within project)")
    description: str = Field(..., description="Skill description")
    file_index: list[str] = Field(
        ..., description="List of relative file paths in the skill"
    )
    meta: dict[str, Any] = Field(
        ..., description="Custom metadata dictionary"
    )
    created_at: str = Field(..., description="ISO 8601 formatted creation timestamp")
    updated_at: str = Field(..., description="ISO 8601 formatted update timestamp")


class ListSkillsOutput(BaseModel):
    """Response model for listing skills."""

    items: list[Skill] = Field(..., description="List of skills")
    next_cursor: str | None = Field(None, description="Cursor for pagination")
    has_more: bool = Field(..., description="Whether there are more items")


class GetSkillFileResp(BaseModel):
    """Response model for getting a skill file."""

    path: str = Field(..., description="File path")
    mime: str = Field(..., description="MIME type of the file")
    url: str | None = Field(None, description="Presigned URL for downloading the file (present if file is not parseable)")
    content: FileContent | None = Field(None, description="Parsed file content if available (present if file is parseable)")

File: src/client/acontext-ts/src/types/skill.ts
/**
 * Type definitions for skill resources.
 */

import { z } from 'zod';

import { FileContentSchema } from './common';

export const FileInfoSchema = z.object({
  path: z.string(),
  mime: z.string(),
});

export type FileInfo = z.infer<typeof FileInfoSchema>;

export const SkillSchema = z.object({
  id: z.string(),
  user_id: z.string().nullable().optional(),
  name: z.string(),
  description: z.string(),
  disk_id: z.string(),
  file_index: z.array(FileInfoSchema),
  meta: z.record(z.string(), z.unknown()).nullable(),
  created_at: z.string(),
  updated_at: z.string(),
});

export type Skill = z.infer<typeof SkillSchema>;

export const SkillCatalogItemSchema = z.object({
  name: z.string(),
  description: z.string(),
});

export type SkillCatalogItem = z.infer<typeof SkillCatalogItemSchema>;

export const ListSkillsOutputSchema = z.object({
  items: z.array(SkillCatalogItemSchema),
  next_cursor: z.string().nullable().optional(),
  has_more: z.boolean(),
});

export type ListSkillsOutput = z.infer<typeof ListSkillsOutputSchema>;

export const GetSkillFileRespSchema = z.object({
  path: z.string(),
  mime: z.string(),
  url: z.string().nullable().optional(),
  content: FileContentSchema.nullable().optional(),
});

export type GetSkillFileResp = z.infer<typeof GetSkillFileRespSchema>;

export const DownloadSkillToSandboxRespSchema = z.object({
  success: z.boolean(),
  dir_path: z.string(),
  name: z.string(),
  description: z.string(),
});

export type DownloadSkillToSandboxResp = z.infer<typeof DownloadSkillToSandboxRespSchema>;

export const DownloadSkillRespSchema = z.object({
  name: z.string(),
  description: z.string(),
  dirPath: z.string(),
  files: z.array(z.string()),
});

export type DownloadSkillResp = z.infer<typeof DownloadSkillRespSchema>;

